ocp_resources/resource.py

```diff
@@ -405,7 +405,7 @@ def __init__(
             raise ValueError("config_file must be a string")
 
         self.config_dict = config_dict or {}
-        self.context = context
+        self.context = context if context else None
```

Comment:
This is not what I asked for, leave it here as it was.
every cal to `get_client` should be with `context=self.context if self.context else None`
Also make the change in all the places that you change the typing.